Assets/MixedRealityToolkit/Utilities/CameraCache.cs

```diff
@@ -1,49 +1,84 @@
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
+using Microsoft.MixedReality.Toolkit.CameraSystem;
 using UnityEngine;
 
 namespace Microsoft.MixedReality.Toolkit.Utilities
 {
     /// <summary>
-    /// The purpose of this class is to provide a cached reference to the main camera. Calling Camera.main
-    /// executes a FindByTag on the scene, which will get worse and worse with more tagged objects.
+    /// The purpose of this class is to provide a cached reference to the main camera.
+    /// If the camera system is enabled, the camera cache will return a reference to that system's camera.
+    /// Otherwise a default camera will be created.
     /// </summary>
     public static class CameraCache
     {
         private static Camera cachedCamera;
 
         /// <summary>
-        /// Returns a cached reference to the main camera and uses Camera.main if it hasn't been cached yet.
+        /// Returns a cached reference to the main camera. Uses the camera system's main camera, if available. Otherwise finds or creates a main camera.
         /// </summary>
         public static Camera Main
         {
             get
             {
-                if (cachedCamera != null)
+                // If the camera system is enabled, use the main camera.
+                // This allows developers to control exactly which camera will be returned by the cache.
+                if (MixedRealityToolkit.IsCameraSystemEnabled)
                 {
-                    if (cachedCamera.gameObject.activeInHierarchy)
-                    {   // If the cached camera is active, return it
-                        // Otherwise, our playspace may have been disabled
-                        // We'll have to search for the next available
-                        return cachedCamera;
-                    }
+                    return MixedRealityToolkit.CameraSystem.Main;
                 }
 
-                // If the cached camera is null, search for main
-                var mainCamera = Camera.main;
+                // Otherwise, return a default camera
+                return FindOrCreateDefaultMainCamera();
+            }
+        }
 
-                if (mainCamera == null)
-                {   // If no main camera was found, create it now
-                    Debug.LogWarning("No main camera found. The Mixed Reality Toolkit requires at least one camera in the scene. One will be generated now.");
-                    mainCamera = new GameObject("Main Camera", typeof(Camera)) { tag = "MainCamera" }.GetComponent<Camera>();
+        /// <summary>
+        /// Returns true if cached camera exists and is enabled.
+        /// </summary>
+        public static bool MainExists
+        {
+            get
+            {
+                if (MixedRealityToolkit.IsCameraSystemEnabled)
+                {   // The camera system is guaranteed to create a camera.
+                    return true;
```

Comment:
I think we should have a more robust check for this compared to always assuming it will be true. I'm not sure if the long term plans are for MixedRealityCameraSystems to be used only in wmr experiences. We may as well add an additional flag in MixedRealityToolkit to check for this compared to building out the assumption that IsCameraSystemEnabled == Camera exists

Reply:
I think it would remove a lot of red tape to make this a requirement of the service interface. The Main property is simply expected to deliver a camera if the service is enabled. Not enabling the service is equivalent to saying 'I don't want MRTK to create / manage a camera for me.'

There are still questions about whether it should be the same camera every time, or how to deal with multiple cameras, or whether the main camera is the one rendering, etc.